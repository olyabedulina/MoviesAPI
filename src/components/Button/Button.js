import React from 'react'
import PropTypes from 'prop-types'

import CM from './styles.pcss'

const Button = ({
    className = '',
    kind = 'main',
    children,
    onClick = Function.prototype
}) => {
    function handleClick() {
        onClick();
    }

    return <button
        className={`${className} ${CM.button} ${CM["kind-" + kind] || ''}`}
        type="button"
        onClick={handleClick}>
            {children}
        </button>
}

Button.propTypes = {
    className: PropTypes.string,
    kind: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

export default Button