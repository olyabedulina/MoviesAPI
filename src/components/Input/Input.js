import React, { useState } from 'react'
import PropTypes from 'prop-types'

import CM from './styles.pcss'

const Input = ({
    className = '',
    kind = '',
    type = 'text',
    value = '',
    placeholder = '',
    children,
    onClick = Function.prototype
}) => {
    const [inputValue, setinputValue] = useState(value)

    function handleInputChange(event) {
        setinputValue(event.target.value)
    }

    function handleKeyDown(event) {
        if (event.key === 'Enter') {
            onClick();
        }
    }

    return <input
        className={`${className} ${CM.input} ${CM["kind-" + kind] || ''}`}
        type={type}
        value={inputValue}
        placeholder={placeholder}
        onChange={handleInputChange}
        onKeyDown={handleKeyDown}>
            {children}
        </input>
}

Input.propTypes = {
    className: PropTypes.string,
    kind: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string.isRequired,
    placeholder: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func
};

export default Input