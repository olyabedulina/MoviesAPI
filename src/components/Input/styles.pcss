.input {
  box-sizing: border-box;
  display: inline-block;
  border: 0;
  border-radius: 6px;
  outline: none;
  width: 100%;
  padding: 16px 20px;
  margin: 0;
  font-size: 20px;
  line-height: 20px;
  background: rgba(255, 255, 255, 0.1);
  color: #fff;
}