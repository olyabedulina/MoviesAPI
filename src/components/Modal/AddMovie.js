import React, { useState } from 'react'
import PropTypes from 'prop-types'

import Input from '../Input'
import MultiSelect from '../MultiSelect'
import Button from '../Button'

import CM from './styles.pcss'
import EditMovie from "./EditMovie";

const AddMovie = ({
    children,
    genres,
    selectedGenresArray = []
}) => {

    const [selectedGenres, setSelectedGenres] = useState(selectedGenresArray)

    function handleSelectedGenresChange(nextSelectedGenres) {
        setSelectedGenres(nextSelectedGenres)
    }

    return <>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <label className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Title</span>
                    <Input
                        className={CM.fieldInput}
                        placeholder='Title here'
                        type='text'
                        value=''
                    />
                </label>
            </div>
        </li>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <label className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Release Date</span>
                    <Input
                        className={CM.fieldInput}
                        placeholder='Select Date'
                        type='date'
                        value=''
                    />
                </label>
            </div>
        </li>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <label className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Movie URL</span>
                    <Input
                        className={CM.fieldInput}
                        placeholder='Movie URL here'
                        type='text'
                        value=''
                    />
                </label>
            </div>
        </li>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <div className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Movie URL</span>
                    <MultiSelect
                        placeholder="Select genre"
                        items={genres}
                        selectedItems={selectedGenres}
                        onChange={handleSelectedGenresChange}
                    />
                </div>
            </div>
        </li>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <label className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Overview</span>
                    <Input
                        className={CM.fieldInput}
                        placeholder='Overview here'
                        type='text'
                        value=''
                    />
                </label>
            </div>
        </li>
        <li className={CM.modalContainerItem}>
            <div className={CM.field}>
                <label className={CM.fieldLabel}>
                    <span className={CM.fieldLabelText}>Runtime</span>
                    <Input
                        className={CM.fieldInput}
                        placeholder='Runtime here'
                        type='text'
                        value=''
                    />
                </label>
            </div>
        </li>
        <li className={CM.modalFooter}>
            <Button
                kind='alt'
                className={CM.modalFooterButton}>
                Reset
            </Button>
            <Button
                kind='main'
                className={CM.modalFooterButton}>
                Submit
            </Button>
        </li>
    </>
}

AddMovie.propTypes = {
    selectedGenresArray: PropTypes.shape({
        id: PropTypes.string,
        src: PropTypes.string,
        title: PropTypes.string,
        releaseDate: PropTypes.number,
        genre: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.string,
                name: PropTypes.string,
            })
        ),
        rating: PropTypes.string,
        movieDuration: PropTypes.shape({
            timing: PropTypes.number,
            units: PropTypes.string
        }),
        url: PropTypes.string,
        description: PropTypes.string
    }),
    genres: PropTypes.array,
    children: PropTypes.node
};

export default AddMovie