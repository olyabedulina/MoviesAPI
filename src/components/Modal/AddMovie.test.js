import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'

import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

import AddMovie from "./AddMovie";
import Button from "../Button";
import Input from "../Input";

describe('Components :: Add Movie unit tests:', () => {

    beforeEach(() => {
        let store = mockStore({ movieList: [ { A: 'A'}, { B: 'B'}] });

        render(
            <Provider store={store}>
                <AddMovie />
            </Provider>
        );
    });

    it('Find title input by placeholder', () => {
        expect(screen.getByPlaceholderText('Title here')).toBeInTheDocument();
    });

    it('Find title input by placeholder and check to be empty', () => {
        expect(screen.getByPlaceholderText('Title here')).toBeEmptyDOMElement();
    });

    it('Find date input by placeholder', () => {
        expect(screen.getByPlaceholderText('Select Date')).toBeInTheDocument();
    });

    it('Find formButtons by test id', () => {
        expect(screen.getByTestId('formButtons')).toBeInTheDocument();
    });

    it('Find Select genre by text', () => {
        expect(screen.getByText(/Select genre/i)).toBeInTheDocument();
    });

    it('Find Runtime by label text', () => {
        expect(screen.getByLabelText('Runtime')).toBeVisible();
    });

    it('Snapshot Test :: "AddMovie" component', () => {
        let store = mockStore({ movieList: [ { A: 'A'}, { B: 'B'}] });

        const actual = renderer
            .create(<Provider store={store}>
                <AddMovie />
            </Provider>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Snapshot Test :: "AddMovie" component', () => {
        let store = mockStore({ movieList: [ { B: 'B'}, { C: 'C'}] });

        const actual = renderer
            .create(<Provider store={store}>
                <AddMovie>foo</AddMovie>
            </Provider>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Snapshot Test :: "AddMovie" component', () => {
        let store = mockStore({ movieList: [ { B: 'B'}, { C: 'C'}] });

        const actual = renderer
            .create(<Provider store={store}>
                <AddMovie>
                    <Button kind="aux" className="buttonAddMovie">+ Add Movie</Button>
                </AddMovie>
            </Provider>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Snapshot Test :: "AddMovie" component', () => {
        let store = mockStore({ movieList: [ { B: 'B'}, { C: 'C'}], movieId: '1'});

        const actual = renderer
            .create(<Provider store={store}>
                <AddMovie>
                    <input type="text" className="Type text" />
                    <Button kind="aux" className="buttonAddMovie">Submit data</Button>
                </AddMovie>
            </Provider>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

    it('Snapshot Test :: "AddMovie" component', () => {
        let store = mockStore({ movieList: [ { B: 'B'}, { C: 'C'}], movieId: '1'});

        const actual = renderer
            .create(<Provider store={store}>
                <AddMovie>
                    <form className="form">
                        <Input
                            className="formInput"
                            placeholder="What do you want to watch?"
                            type="text"
                            value="0"
                            onChange={() => {}}
                        />
                    </form>
                </AddMovie>
            </Provider>)
            .toJSON();
        expect(actual).toMatchSnapshot();
    });

});
