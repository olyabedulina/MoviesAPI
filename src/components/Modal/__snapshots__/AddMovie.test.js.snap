// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components :: Add Movie unit tests: Snapshot Test :: "AddMovie" component 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="modalContainerItem"
    data-testid="modalContainerItem"
  >
    <h3>
      Add movie
    </h3>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <label>
        <span
          className="fieldLabelText"
        >
          Title
        </span>
        <input
          className=" input "
          id="title"
          name="title"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Title here"
          type="text"
          value=""
        />
      </label>
    </div>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <label>
        <span
          className="fieldLabelText"
        >
          Release Date
        </span>
        <input
          className=" input "
          id="selectDate"
          name="selectDate"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Select Date"
          type="date"
          value=""
        />
      </label>
    </div>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <label>
        <span
          className="fieldLabelText"
        >
          Movie URL
        </span>
        <input
          className=" input "
          id="movieURL"
          name="movieURL"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Movie URL here"
          type="text"
          value=""
        />
      </label>
    </div>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <div>
        <span
          className="fieldLabelText"
        >
          Genre
        </span>
        <div
          className="  dropdown "
        >
          <div
            className="dropdownInput"
          >
            <div
              className="dropdownPlaceholder"
            >
              Select genre
            </div>
            <div
              className="dropdownInputButton dropdownInputButtonUp"
              onClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <label>
        <span
          className="fieldLabelText"
        >
          Overview
        </span>
        <input
          className=" input "
          id="overview"
          name="overview"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Overview here"
          type="text"
          value=""
        />
      </label>
    </div>
  </div>
  <div
    className="modalContainerItem"
  >
    <div
      className="field"
    >
      <label>
        <span
          className="fieldLabelText"
        >
          Runtime
        </span>
        <input
          className=" input "
          id="runtime"
          name="runtime"
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Runtime here"
          type="text"
          value=""
        />
      </label>
    </div>
  </div>
  <div
    className="modalFooter"
    data-testid="formButtons"
  >
    <button
      className="modalFooterButton button kind-alt"
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <button
      className="modalFooterButton button kind-main"
      onClick={[Function]}
      type="submit"
    >
      Submit
    </button>
  </div>
</form>
`;
